library/vector.lisp

```diff
@@ -46,9 +46,8 @@
   ;; Vector
   ;;
 
-  (repr :transparent)
-  (define-type (Vector :a)
-    (%Vector Lisp-Object))
+  (repr :native (cl:array cl:t (cl:*)))
```

Comment:
see other comment, but this is equivalent to `(cl:vector cl:t)`

We would get speed benefits if we could guarantee this to be a `simple-array` (no displacement, no fill pointers, etc)

Reply:
We need fill pointers for push and pop. Vectors will never be displaced, but they can be overlapped by slices.